x/ccv/provider/keeper/throttle.go

```diff
@@ -29,9 +29,20 @@ func (k Keeper) HandlePendingSlashPackets(ctx sdktypes.Context) {
 	// Iterate through ordered (by received time) slash packet entries from any consumer chain
 	k.IteratePendingSlashPacketEntries(ctx, func(entry providertypes.SlashPacketEntry) (stop bool) {
 
+		// Obtain validator from the provider's consensus address.
+		// Note: if validator is not found or unbonded, this will be handled appropriately in HandleSlashPacket
+		val, found := k.stakingKeeper.GetValidatorByConsAddr(ctx, entry.ProviderValConsAddr)
+
 		// Obtain the validator power relevant to the slash packet that's about to be handled
 		// (this power will be removed via jailing or tombstoning)
-		valPower := k.stakingKeeper.GetLastValidatorPower(ctx, entry.ValAddr)
+		var valPower int64
+		if !found || val.IsJailed() {
+			// If validator is not found, or found but jailed, it's power is 0. This path is explicitly defined since the
+			// staking keeper's LastValidatorPower values are not updated till the staking keeper's endblocker.
+			valPower = 0
+		} else {
+			valPower = k.stakingKeeper.GetLastValidatorPower(ctx, val.GetOperator())
+		}
```

Comment:
I might be missing something but doesn't this change allow validator to be slashed all the way to 0 in a single block?

If a consumer sends many slash packets for the same validator?

_______________________________________________________________________________________________


Besides that, I don't think this addressing the problem in this comment

https://github.com/cosmos/interchain-security/pull/462#issuecomment-1339845142

To state the problem more clearly:

We want it to be possible to set the slashFraction or other params so that the throttle is 'invisible'. Ie. the behavior of the system is exactly as in the current [spec](https://github.com/cosmos/ibc/blob/main/spec/app/ics-028-cross-chain-validation/system_model_and_properties.md#consumer-initiated-slashing),
This means, we do not want any slash packets to be held up/delayed (still talking about when fraction=1, replenishPeriod = 1 second ect).

Consider the following

Imagine there are n validators total on the provider, with k<n active. Imagine the actives are v1,v2..vk and the inactives are vk+1,..vn

A consumer sends slash packets for every single provider validator v1..,vn and they are all delivered to the provider in the same block.

It is possible, that all of the slashes for the active validators v1..vk are handled first. This will set the meter to 0 so the slashes for vk+1,.. vn will be blocked.

Reply:
With this design we can slash up to but not including 100% of the "initial val set" instantaneously when the replenish fraction is 1. If we want to include 100%, its a one line inequality change. I like 100% not being allowed, but I'm not super attached to that idea

Reply:
The change we discussed today is included in https://github.com/cosmos/interchain-security/pull/565/commits/777a3f6f5a771aabf73d1a591c8de7ae413f4be1, thank you for the efforts and feedback. Even with disagreements at times, the dialogue ended up useful and productive! It will be cool to get this feature properly specced and thoroughly tested before prod

Reply:
and before I forget, 

> I might be missing something but doesn't this change allow validator to be slashed all the way to 0 in a single block?

Yes this is possible with or without slash packet throttling. Does https://github.com/cosmos/interchain-security/issues/544 address your concern? If not, might be useful to add onto that issue or create a new one